src/Drupal8/Rector/Deprecation/EntityManagerRector.php

```diff
@@ -145,17 +148,18 @@ public function findInstanceByNameInAssign(Node\Expr\Assign $assign, string $cla
         $depth = 0;
 
         // Should the expression be the class we are looking for and the name is the one we are looking for, we can return early.
-        if ($node instanceof $class && $this->getName($node->name) === $name) {
+        if ($node instanceof Node && $node instanceof $class && $this->getName($node->name) === $name) {
             $node->setAttribute(self::class, $depth);
 
             return $node;
         }
 
         // Find the relevant class with name in the chain.
-        while (isset($node->var) && !($node->var instanceof $class && $this->getName($node->var->name) !== $name)) {
+        while (isset($node->var) && !($node->var instanceof $class && isset($node->var->name) && $this->getName($node->var->name) !== $name)) {
             $node = $node->var;
             ++$depth;
-            if ($node instanceof $class && $this->getName($node->name)) {
+
+            if ($node instanceof Node && $node instanceof $class && isset($node->name) && $this->getName($node->name)) {
```

Comment:
again, feels weird that we have to `instanceof` something which shouldn't be nullable and is `\PhpParser\Node\Expr`

And we can call `$this->getName($node)`, that's what the name resolver is for.